core/unix/os_exports.h

```diff
@@ -99,16 +105,22 @@
 #    ifdef X64
 #        define SEG_TLS DR_REG_TPIDRRO_EL0   /* DR_REG_TPIDRURO, but we can't use it */
 #        define LIB_SEG_TLS DR_REG_TPIDR_EL0 /* DR_REG_TPIDRURW, libc+loader tls */
+#        define STR_SEG "tpidruro"
+#        define STR_LIB_SEG "tpidrurw"
 #    else
 #        define SEG_TLS                                                     \
             DR_REG_TPIDRURW /* not restored by older kernel => we can't use \
                              */
 #        define LIB_SEG_TLS DR_REG_TPIDRURO /* libc+loader tls */
-#    endif                                  /* 64/32-bit */
+#        define STR_SEG "tpidrurw"
+#        define STR_LIB_SEG "tpidruro"
+#    endif /* 64/32-bit */
 #elif defined(RISCV64)
 /* FIXME i#3544: Not used on RISC-V, so set to invalid. Check if this is true. */
 #    define SEG_TLS DR_REG_INVALID
 #    define LIB_SEG_TLS DR_REG_TP
+#    define STR_SEG "tp"
+#    define STR_LIB_SEG "s11"
```

Comment:
Shouldn't this be "tp" based on the above definition for LIB_SEG_TLS? And STR_SEG should be "invalid" or something?

Reply:
Yes, it should be. Plus I should fix the `read_thread_register()` and `get_segment_base()` to adapt to this.